layers/+lang/lean/packages.el

```diff
@@ -0,0 +1,41 @@
+;;; packages.el --- Lean Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+(setq lean-packages '(mmm-mode 
+                      f 
+                      dash-functional
+                      (lean-mode :location (recipe
+                                            :fetcher github
+                                            :repo leanprover/lean
+                                            :files ("src/emacs/*.el")))))
+(defun lean/init-f ())
+
+(defun lean/init-dash-functional ())
+
+(defun lean/init-mmm-mode ())
+
+(defun lean/init-lean-mode ()
+  (use-package lean-mode
+    :defer t
+    :config
+    (spacemacs/set-leader-keys-for-major-mode 'lean-mode
+      "L" 'lean-std-exe
+      "R" 'lean-server-restart-process
+      "d" 'lean-eldoc-documentation-function
```

Comment:
Just pointing out it'd be good to take look at [CONVENTIONS doc](https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.org#help-or-documentation) which will help you write better layer. :smile: 
So I'm unsure about this keybinding, if it shows documentation, it should be under `SPC m h h` (since there's eldoc, I'm unclear as to what it does so use your best judgment then)